tests/rptest/tests/redpanda_test.py

```diff
@@ -50,11 +55,27 @@ def __init__(self,
             else:
                 num_brokers = 1
 
+        cloud_storage_access_key = test_context.globals.get(
+            self.GLOBAL_S3_ACCESS_KEY, None)
+        cloud_storage_secret_key = test_context.globals.get(
+            self.GLOBAL_S3_SECRET_KEY, None)
+
+        # Enable S3 if AWS creds were given at globals
+        if cloud_storage_access_key is not None and cloud_storage_secret_key is not None:
+            if self.si_settings is not None:
+                self.si_settings.cloud_storage_access_key = cloud_storage_access_key
+                self.si_settings.cloud_storage_secret_key = cloud_storage_secret_key
+                self.si_settings.endpoint_url = None  # None so boto auto-gens the endpoint url
```

Comment:
IIRC @jcsp recommended this.. Just curious; why not use explicit URL in both cases (less code, no chance for diverging)?

Reply:
> IIRC @jcsp recommended this.. Just curious; why not use explicit URL in both cases (less code, no chance for diverging)?

I see your point but I think using None is better because we won't have to worry about boto changes in the future. Just let boto auto-gen the url.

Reply:
Using None here also exercises redpanda's S3 client's inference of AWS S3 urls (the thing that earlier on I didn't think existed but it turns out does), which is a good thing.  End users in S3 probably won't bother overriding this, so I think it's appropriate for the tests to run redpanda that way too.

Reply:
Ah.. I figured we didn't have inference, and boto3/s3 URI API is probably pretty stable, but yeah.. coverage is good.